Core/Versioning/KspVersionCriteria.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CKAN.Versioning
+{
+    public class KspVersionCriteria
```

Comment:
The class should also *behave* like a set and only keep a single copy of duplicate `KspVersion` objects. This should be easy as `KspVersion` implements `IEquatable<KspVersion>`.